File: cpp/problem94.cpp
#include "TreeNode.h"
#include <iostream>
#include <vector>
using namespace std;

class Solution {
public:
    vector<int> inorderTraversal(TreeNode* root) {
        vector<int> res;
        rec(root, res);
        return res;
    }
    void rec(TreeNode* node, vector<int> &res){
        if (node){
            rec(node->left, res);
            res.push_back(node->val);
            rec(node->right,res);
        }
    }
};

File: python/problem94.py
import TreeNode


class Solution:
    def inorderTraversal(self, root: TreeNode) -> List[int]:
        l = []

        def rec(root):
            if root.left:
                rec(root.left)
            l.append(root.val)
            if root.right:
                rec(root.right)

        if root:
            rec(root)
        return l
